src/permutat.cc

```diff
@@ -826,14 +826,7 @@ Obj PowIntPerm(Obj opL, Obj opR)
         return opL;
 
     /* permutations do not act on negative integers                        */
-    img = INT_INTOBJ( opL );
-    if ( img <= 0 ) {
-        opL = ErrorReturnObj(
-            "Perm. Operations: <point> must be a positive integer (not %d)",
-            (Int)img, 0L,
-            "you can replace <point> via 'return <point>;'" );
-        return POW( opL, opR );
-    }
+    img = GetPositiveSmallInt("PermImage", opL, "point");
```

Comment:
There is a reason why I did not just replace the above with `RequirePositiveSmallInt` in my earlier rounds: The error message after this change is somewhat misleading: `opL` only needs to be a positive int, not necessarily small; the large int case simply is handled before. Of course one can argue that one does not care about this very minor regression, but one should at least be aware of it (and perhaps add a comment pointing this out?)